_overviews/scala3-book/taste-control-structures.md

```diff
@@ -0,0 +1,250 @@
+---
+title: Control Structures
+type: section
+description: This section demonstrates Scala 3 control structures.
+num: 8
+previous-page: taste-vars-data-types
+next-page: taste-modeling
+---
+
+
+Scala has the programming language control structures you find in other languages, and also has powerful `for` expressions and `match` expressions:
+
+- `if`/`else`
+- `for` loops and expressions
+- `match` expressions
+- `while` loops
+- `try`/`catch`
+
+These structures are demonstrated in the following examples.
+
+
+
+## `if`/`else`
+
+Scala’s `if`/`else` control structure is similar to other languages:
+
+```scala
+if x < 0 then
+  println("negative")
+else if x == 0
+  println("zero")
+else
+  println("positive")
+```
+
+Note that this really is an _expression_ — not a _statement_. This means that it returns a value, so you can assign the result to a variable:
+
+```scala
+val x = if a < b then a else b
+```
+
+As you’ll see throughout this book, _all_ Scala control structures can be used as expressions.
+
+>An expression returns a result, while a statement does not. Statements are typically used for their side-effects, such as using `println` to print to the console.
+
+
+
+## `for` loops and expressions
+
+The `for` keyword is used to create a `for` loop. This example shows how to print every element in a `List`:
+
+```scala
+val ints = List(1, 2, 3, 4, 5)
+
+for i <- ints do println(i)
+```
+
+The code `i <- ints` is referred to as a _generator_, and if you leave the parentheses off of the generator, the `do` keyword is required before the code that follows it. Otherwise you can write the code like this:
+
+```scala
+for (i <- ints) println(i)
+```
```

Comment:
It is unclear what parentheses are referred to if you have never seen Scala 2 code. So I would rather say:

```suggestion
The code `i <- ints` is referred to as a _generator_, and the code that follows the `do` keyword is the _body_ of the loop.

The old syntax for this control structure was:

```scala
for (i <- ints) println(i)
```
```